plugins/repository-s3/src/main/java/org/elasticsearch/repositories/s3/S3Service.java

```diff
@@ -157,6 +157,10 @@ AmazonS3 buildClient(final S3ClientSettings clientSettings) {
         if (clientSettings.pathStyleAccess) {
             builder.enablePathStyleAccess();
         }
+
+        if (clientSettings.disableChunkedEncoding) {
+            builder.withChunkedEncodingDisabled(true);
```

Comment:
Slight preference for this, avoiding the `boolean` parameter:

```suggestion
            builder.disableChunkedEncoding();
```